src/synapse.py

```diff
@@ -146,51 +146,70 @@ def execute_migrate_config(self, container: ops.Container) -> None:
                 "Migrate config failed, please review your charm configuration"
             )
 
-    def check_server_name(self, container: ops.Container) -> None:
-        """Check server_name.
-
-        Check if server_name of the state has been modified in relation to the configuration file.
+    def get_configuration_field(
```

Comment:
Could this be a private function?

Reply:
Actually no because Synapse API needs it to get the registration_shared_secret

Reply:
ok, then should this method be on the `Synapse` class? It could just take the container as an argument and then we have less coupling between Synapse and Synapse API

Reply:
I have an idea: I created a public method called get_registration_shared_secret that will be called in the charm only. Then _get_configuration_field is private now.